docs/reference/api/connectivity/bluetooth/GattClient.md

```diff
@@ -1,6 +1,26 @@
 ## GattClient
 
-[Add description here.]
+GattClient defines procedures required for interacting with a remote GattServer.
+
+#### Discovery procedures
+
+A GattServer hosts a fixed set of services. These services are a logical composition of characteristics that may be discovered, read, or written, and can broadcast their state to a connected client. These characteristics may also contain metainformation and named characteristic descriptors. A characteristic descriptor may indicate the unit used for a characteristic value, describe the characteristic purpose in a textual form or allow a client to register for update notifications for the characteristic value.
+
+Prior to any interaction with a server characteristic, a GattClient discovers the layout of the services and characteristics present on the server.
+
+The layout of the descriptors of a characteristic may also be issued as an extra discovery step.
+
+#### Attribute manipulation
+
+As a result of the discovery process, the client can start interacting with the characteristic discovered. Depending on the characteristic properties (acquired during discovery), a client can read or write the value of a given characteristic.
+
+Mbed BLE abstracts read and write operations to offer a single API that can be used to read or write characteristic values. The application code does not have to handle the necessary fragmentation/reassembly process if the attribute value to be transported cannot fit in a single data packet.
+
+#### Server Initiated events
+
+A characteristic can notify or indicate a property set, and a client may register to this notification or indication from the characteristic. When the server updates the characteristic value, the server can also forward the new value to the registered clients. The notification/indication mechanism prevents polling from the client, and therefore, minimizes the transactions between a client and a server.
+
+Registration is made by writing the Client Characteristic Configuration Descriptor, which is present in the characteristic if the notify or indicate properties are set. The client discovers that descriptor if it intends to register to server initiated events.
```

Comment:
Clarification needed on this 1st sentence.  At present, it sounds like 'writing to' the client characteristic configuration descriptor' (also why capitals here?), or do you mean writing the descriptor itself? And if so, writing it where?  The 2nd sentence is more confusing.  It sounds like you're trying to say that if the notify/indicate properties are set, then this descriptor is present in the characteristic, and the client can find this descriptor and use it to register by writing something somewhere.  Please clarify and I can restructure the sentences a bit.

Reply:
It's capitalised to indicate a specific term often abbreviated. Give me a moment to rework this.

Reply:
hopefully that's clearer